packages/fx-core/src/common/local/taskDefinition.ts

```diff
@@ -0,0 +1,242 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+"use strict";
+
+import { FolderName, ProgrammingLanguage, npmInstallCommand } from "./constants";
+import path from "path";
+import { isWindows } from "../deps-checker/util/system";
+
+export interface ITaskDefinition {
+  name: string;
+  command: string;
+  cwd: string | undefined;
+  isBackground: boolean;
+  isShell: boolean;
```

Comment:
Better naming to `needShell` and `needCmd`, and putting them into `execOptions`:
```
  execOptions?: { needShell: boolean; needCmd?: boolean };
```

Since `cmd` is bound with `shell === true`.